indico/modules/events/abstracts/forms.py

```diff
@@ -474,7 +474,10 @@ def __init__(self, *args, **kwargs):
         if abstracts_settings.get(self.event, 'contrib_type_required'):
             inject_validators(self, 'submitted_contrib_type', [DataRequired()])
         super(AbstractForm, self).__init__(*args, **kwargs)
-        self.submitted_contrib_type.query = self.event.contribution_types
+        if self.event.can_manage(session.user):
+            self.submitted_contrib_type.query = self.event.contribution_types
+        else:
+            self.submitted_contrib_type.query = self.event.contribution_types.filter_by(is_private=False)
```

Comment:
This is not covering that case where you are editing an existing abstract that has a private type. I think this might do the job:

```python
criteria = [~ContributionType.is_private]
if self.abstract and self.abstract.submitted_contrib_type:
    criteria.append(ContributionType.id == self.abstract.submitted_contrib_type.id)
self.submitted_contrib_type.query = self.event.contribution_types.filter(db.or_(*criteria))
```